pkg/component/component_test.go

```diff
@@ -1513,15 +1502,6 @@ func TestToComponents(t *testing.T) {
 						BinaryPath: filepath.Join("..", "..", "specs", "apm-server"),
 					},
 					Units: []Unit{
-						{
```

Comment:
This is the only change in the tests that is functional rather than just switching how the shipper is enabled: in the old model, where outputs could optionally enable any set of available shippers, it made sense to fall back to a non-shipper output if an input didn't support one of them. Since shipper processing is now a boolean all-or-nothing flag on the output, we no longer want to silently substitute a non-shipper output if it isn't available. In this case, the configured input doesn't support any shippers, and so this output unit should no longer be created; instead, an error is returned in the created component indicating that the shipper isn't supported by the input.

Reply:
I don't have the right functional context for this, even if I prefer the all-or-nothing approach instead of magic behind the scenes.
From a testing perspective, I was expecting to see a new error assertion somewhere... Did I miss it? Or we silently do not output anything?

Reply:
No, you're right, thank you! The test harness didn't verify the lack of errors, so until you pointed it out I forgot to add it. I've now made it a test failure for an error to appear on a component where one wasn't expected, and added the new expected error ("this input does not support shipper outputs") on this test case.